manual/parts/glossary.tex

```diff
@@ -0,0 +1,116 @@
+}
+
+\newglossaryentry{guard}{
+    name=Guard,
+    description={Guard of a CNode capability. From the user’s perspetive the CSpace of a thread is organised as a guarded page table. The kernel will resolve user capability pointers into internal capability slot pointers. The guard of one link/edge in the CSpace graph defines a sequence of bits that will be stripped from the user-level capability pointer before resolving resumes at the next CNode}
+}
+
+\newglossaryentry{iommu}{
+    name=IOMMU,
+    description={Input–Output Memory Management Unit. Applies virtual address translation and memory protection to DMA capable I/O devices}
+}
+
+\newglossaryentry{iopagetable}{
+    name=IOPageTable,
+    description={This object represents a node in the multilevel page-table structure used by IOMMU hardware to translate hardware memory accesses}
+}
+
+\newglossaryentry{iospace}{
+    name=IOSpace,
+    description={This object represents the address space associated with a hardware device. It represents the right to modify a device’s address space. See \autoref{ch:io}}
+}
+
+\newglossaryentry{ipc}{
+    name=IPC,
+    description={Inter Process Communication is facilitated by endpoints, which act as general communication ports. Invocations on endpoint objects are used to send and receive messages}
+}
+
+\newglossaryentry{irqcontrol}{
+    name=IRQControl,
+    description={A single capability from which IRQHandler capabilities to all IRQ numbers in the system can be derived. This capability can be moved between CSpaces and CSlots but cannot be duplicated. Revoking this capability removes all IRQHandlers}
+}
+
+\newglossaryentry{irqhandler}{
+    name=IRQHandler,
+    description={Capabilities that represent the ability of a thread to handle a certain interrupt. See \autoref{ch:io}}
+}
+
+\newglossaryentry{notificationobject}{
+    name=Notification Object,
+    description={A word-size array of flags that provides a non-blocking signalling mechanism similar to a binary semaphore. Operations are signalling a subset of flags in a single operation, polling to check any flags, and blocking until any are signalled. Notification capabilities can be signal-only or wait-only}
+}
+
+\newglossaryentry{replyobject}{
+    name=Reply Object,
+    description={A reply object is a vessel for tracking reply messages, used to send a reply message and
+    wake up the caller}
+}
+
+\newglossaryentry{schedulingcontext}{
+    name=Scheduling Context,
+    description={(MCS only) An abstraction of CPU execution time}
+}
+
+\newglossaryentry{tcb}{
+    name=TCB,
+    description={Thread Control Block. The kernel object that stores management data for threads, such as the thread's CSpace, VSpace, thread state, or user registers}
+}
+
+\newglossaryentry{untypedmemory}{
+    name=Untyped Memory,
+    description={The foundation of memory allocation in the seL4 kernel. See \autoref{sec:kernmemalloc}}
```

Comment:
```suggestion
    name=Untyped Memory,
    description={Memory that can be used to create kernel objects or user-accessible frames. The foundation of memory allocation in the seL4 kernel. See \autoref{sec:kernmemalloc}}
```

Reply:
Strictly speaking it can only be used to create kernel objects, except if Page objects are not kernel objects, in which case that should be mentioned explicitly (instead of frame). I would go for:

"The foundation of kernel object creation and memory allocation in the seL4 kernel."

Reply:
It's correct that frames are also kernel objects, the difference I was trying to point out is that these are directly writeable by user space, and the rest are not. Happy to tweak that. What I'm not happy with in the "foundation" sentence before was that it doesn't say what you can do with these objects. That untyped memory is the foundation of something doesn't help the reader to understand where to go with that. Maybe we should just point at the retype operation. (I'm happy mentioning the foundation bit, it should just not be the only thing).

Reply:
You're right, the whole user-writeable thing is leading in the wrong direction. Happy to remove that.

Reply:
@lsf37 @Indanz I'm getting a bit bogged down in details :-)
What wording would you like to use for the Description of Untyped Memory?

Reply:
I'd go for

```suggestion
    name=Untyped Memory,
    description={Memory that can be used to create kernel objects via the \apifunc{seL4\_Untyped\_Retype}{untyped_retype} invocation. It is the foundation of memory allocation in the seL4 kernel. See \autoref{sec:kernmemalloc}}
```